### PR TITLE
Add null safety migration annotations.

### PR DESCRIPTION
**Pull request by Hixie**
This adds annotations to prepare the following files for migration:

* rendering/animated_size.dart
* rendering/binding.dart
* rendering/box.dart
* rendering/custom_layout.dart
* rendering/custom_paint.dart
* rendering/debug.dart
* rendering/debug_overflow_indicator.dart
* rendering/editable.dart
* rendering/error.dart
* rendering/flex.dart
* rendering/flow.dart

(The rendering directory, alphabetically up to flow.dart.)

### CODE REVIEW

**Comment by Hixie** [comment]
This brings me back to the days where we were setting up method generics before they were allowed...

**Comment by Hixie** [comment]
cc @goderbauer

File: packages/flutter/lib/src/rendering/box.dart [hunk 1]
```diff
@@ -1294,9 +1294,9 @@ abstract class RenderBox extends RenderObject {
       child.parentData = BoxParentData();
   }
 
-  Map<_IntrinsicDimensionsCacheEntry, double> _cachedIntrinsicDimensions;
+  Map<_IntrinsicDimensionsCacheEntry, double/*!*/> _cachedIntrinsicDimensions;
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/box.dart:1297]: Presumably, `_IntrinsicDimensionsCacheEntry` would also be `/*!*/` ?

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
Looks like a test is failing, though.

File: packages/flutter/lib/src/rendering/custom_paint.dart [hunk 2]
```diff
@@ -675,15 +676,15 @@ class RenderCustomPaint extends RenderProxyBox {
...
+  static List<SemanticsNode/*!*/> _updateSemanticsChildren(
     List<SemanticsNode> oldSemantics,
     List<CustomPainterSemantics> newChildSemantics,
   ) {
     oldSemantics = oldSemantics ?? const <SemanticsNode>[];
     newChildSemantics = newChildSemantics ?? const <CustomPainterSemantics>[];
 
     assert(() {
-      final Map<Key, int> keys = HashMap<Key, int>();
+      final Map<Key/*!*/, int> keys = HashMap<Key/*!*/, int>();
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/rendering/custom_paint.dart:687]: inferred

File: packages/flutter/lib/src/rendering/box.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/rendering/box.dart:1297]: somehow it got that right

**Comment by Hixie** [comment]
Updated as suggested

**Review by goderbauer — APPROVED** [review]
Still LGTM

**Pull request merged**